api/column_family.cc

```diff
@@ -67,21 +67,21 @@ future<> foreach_column_family(http_context& ctx, const sstring& name, function<
 future<json::json_return_type>  get_cf_stats(http_context& ctx, const sstring& name,
         int64_t replica::column_family_stats::*f) {
     return map_reduce_cf(ctx, name, int64_t(0), [f](const replica::column_family& cf) {
-        return cf.get_stats().*f;
+        return cf.rebuild_and_get_stats().*f;
```

Comment:
Maybe get_stats() should do it? The caller shouldn't care about when statistics are updated.

Reply:
I think that we use the method also internally for updating the stats in place, so maybe the mutable overload should just return a reference to the member while the const overload can rebuild it and then return a const&

Reply:
No, overload on const shouldn't have a difference in behavior. We should rename the current get_stats() to get_stale_stats_fast() to make it clear that it's incomplete.

Reply:
How about the following:
```c++
    const table_stats& mutate_stats(std::function<void(table_stats&)> func) {
        func(_stats);
        return _stats;
    }
```

There are 3 call sites on the paxos path that need to update the stats in-place.
I also found out that the metrics return stale gauges.
In the next version I'm about to push, I added `rebuild_statistics_periodic` that updates the stats at most once every second for that.

Let me push it and let's discuss v2.
I think it makes sense overall.